src/portfft/descriptor.hpp

```diff
@@ -741,8 +802,15 @@ class committed_descriptor {
     std::size_t n_kernels = params.lengths.size();
     for (std::size_t i = 0; i < n_kernels; i++) {
       dimensions.push_back(build_w_spec_const<PORTFFT_SUBGROUP_SIZES>(i));
-      dimensions.back().kernels.at(0).twiddles_forward =
-          std::shared_ptr<Scalar>(calculate_twiddles(dimensions.back()), [queue](Scalar* ptr) {
+      dimensions.back().forward_kernels.at(0).twiddles_forward = std::shared_ptr<Scalar>(
+          calculate_twiddles(dimensions.back().level, dimensions.back().forward_kernels), [queue](Scalar* ptr) {
+            if (ptr != nullptr) {
+              sycl::free(ptr, queue);
+            }
+          });
+      // TODO: refactor multi-dimensional fft's such that they can use a single pointer for twiddles.
+      dimensions.back().backward_kernels.at(0).twiddles_forward = std::shared_ptr<Scalar>(
+          calculate_twiddles(dimensions.back().level, dimensions.back().backward_kernels), [queue](Scalar* ptr) {
```

Comment:
Can we use `utils::make_shared` here?

Reply:
just remembered why I hadn't used `utils::make_shared` in the first place, because it does not accept a callable.
I would prefer not to complicate that function in this PR, especially when it needs to call a member function of `committed_descriptor`